design/ceph/object/ceph-rgw-k8s-sa-authentication.md

```diff
@@ -0,0 +1,182 @@
+
+Other settings are similar to the configuration mentioned in [here](/Documentation/Storage-Configuration/Advanced/key-management-system/#kubernetes-based-authentication)
+
+The `Ceph Object Store` CRD will look as :
+
+```yaml
+spec:
+  security:
+    kms:
+      connectionDetails:
+          KMS_PROVIDER: vault
+          VAULT_AGENT_ADDR: http://127.0.0.1:8100
+          VAULT_AGENT_CM: <ceph-object-store>-vault-agent-cm
+          VAULT_BACKEND_PATH: rook
+          VAULT_SECRET_ENGINE: kv
+          VAULT_AUTH_METHOD: kubernetes
+
+---
+kind: ConfigMap
+metadata:
+  name: <ceph-object-store>-vault-agent-cm
+  namespace: rook-ceph
+apiVersion: v1
+data:
+
+  config.hcl: |
+    pid_file = "/home/vault/pidfile"
+
+    auto_auth {
+        method "kubernetes" {
+            mount_path = "auth/kubernetes"
+            config = {
+                role = "rook-ceph"  # role assigned to service account
+            }
+        }
+    }
+
+    cache {
+        use_auto_auth_token = true
+    }
+
+    exit_after_auth = false
+
+    listener "tcp" {
+        address = "127.0.0.1:8100" # port where vault agent listens in the pod
+        tls_disable = "true"
+    }
+
+    vault {
+      address = "http://vault.default:8200" # address of vault server
+      tls_skip_verify = "true"
+    }
+
+  config-init.hcl: |
+    pid_file = "/home/vault/pidfile"
+
+    auto_auth {
+        method "kubernetes" {
+            mount_path = "auth/kubernetes"
+            config = {
+                role = "rook-ceph" # role assigned to service account
+            }
+        }
+    }
+
+    cache {
+        use_auto_auth_token = true
+    }
+
+    listener "tcp" {
+        address = "127.0.0.1:8100" # port where vault agent listens in the pod
+        tls_disable = "true"
+    }
+
+    exit_after_auth = true
+
+    vault {
+      address = "http://vault.default:8200" # address of vault server
+      tls_skip_verify = "true"
+    }
+
+---
+pod:
+  annotations:
+    vault.hashicorp.com/agent-inject: 'true'
+    vault.hashicorp.com/agent-configmap: <ceph-object-store>-vault-agent-cm
+
+```
+
+### TLS authentication
+
+For TLS enabled vault server, the certs can be provided with help of k8s secret and added to `vault.hashicorp.com/tls-secret: <name of k8s secret>` and configure the `vault` block accordingly. The secret will be mounted in the path `/vault/tls/`.
+
+```yaml
+pod:
+  annotations:
+    vault.hashicorp.com/agent-inject: 'true'
+    vault.hashicorp.com/agent-configmap: <ceph-object-store>-vault-agent-cm
+    vault.hashicorp.com/tls-secret: <name of k8s secret>
+    vault.hashicorp.com/ca-cert: '/vault/tls/ca.crt'
+    vault.hashicorp.com/client-cert: '/vault/tls/client.crt'
+    vault.hashicorp.com/client-key: '/vault/tls/client.key'
+
+---
+kind: Secret
+metadata:
+  name: <name of k8s secret>
+  namespace: rook-ceph
+type: Opaque
+data:
+  client.key: <>
+  client.crt: <>
+  ca.crt:     <>
+
```

Comment:
In this scenario, how does RGW know where to get the client ca/cert/key? Does it always load from `/vault/tls`?